crates/vm/src/system/memory/online/access_adapter.rs

```diff
@@ -0,0 +1,453 @@
+use std::{
+    borrow::{Borrow, BorrowMut},
+    io::Cursor,
+};
+
+use openvm_circuit_primitives::{
+    is_less_than::{IsLessThanIo, IsLtSubAir},
+    var_range::SharedVariableRangeCheckerChip,
+    AlignedBorrow, SubAir, TraceSubRowGenerator,
+};
+use openvm_stark_backend::{
+    interaction::InteractionBuilder,
+    p3_air::{Air, BaseAir},
+    p3_field::{FieldAlgebra, PrimeField32},
+    p3_matrix::Matrix,
+    rap::{BaseAirWithPublicValues, PartitionedBaseAir},
+};
+
+use crate::system::memory::{
+    offline_checker::{MemoryBus, AUX_LEN},
+    MemoryAddress,
+};
+
+#[repr(C)]
+#[derive(Debug, Clone, AlignedBorrow)]
+/// The columns of the AccessAdapterAir, indicating one split or merge operation.
+pub(crate) struct AccessAdapterCols<T, const LEN: usize, const NUM_BLOCKS: usize> {
```

Comment:
where was this struct before? if possible keep it where it was since it's part of the AIR